src/ripple/beast/rfc2616.h

```diff
@@ -257,7 +257,10 @@ split(FwdIt first, FwdIt last, Char delim)
         }
         else if (is_lws (*iter))
         {
-            ++iter;
+            e.append (1, ' ');
```

Comment:
Instead of unconditionally adding a space, what do you think about:
```
            if (!e.empty())
                e.append (1, ' ');
```
That way we would only add a space when we are not starting a new token and we could save the `trim_left` call (i.e. replace `trim` with `trim_right`).

Reply:
I'm confused as to how that would work. Remember, we aren't splitting the word only at spaces (e.g. look for `split_comma` below, which splits on commas.

Reply:
I have a better idea, lets get those tests up first so we can visualize the cases that we're discussing!

Reply:
@nbougalis As written, this code collapses all linear white space into a single space. When we add a token to the result, we trim the left and right white space. I'm proposing we change this so when starting a new token we collapse all linear white space into nothing, so we no longer need the left trim.

I.e. there's no reason to add a space to a new token just to trim it away later.

Reply:
Here's a different [implementation](https://gist.github.com/thejohnfreeman/d1e298e0d860a1bf539e6b0178ae05d9) that's shorter, with fewer else-branches, more DRY, and more early exits, but with a `goto`.

Reply:
I appreciate that implementation @thejohnfreeman but I'm slightly allergic to `goto`. It _may_ be OK here, and I think it does help make the implementation a little clearer, but I need to think whether I want to adopt that approach.